homeassistant/components/climate/opentherm_gw.py

```diff
@@ -0,0 +1,187 @@
+"""
+Support for OpenTherm Gateway devices.
+
+For more details about this component, please refer to the documentation at
+http://home-assistant.io/components/climate.opentherm_gw/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.climate import (ClimateDevice, PLATFORM_SCHEMA,
+                                              STATE_IDLE, STATE_HEAT,
+                                              STATE_COOL,
+                                              SUPPORT_TARGET_TEMPERATURE)
+from homeassistant.const import (ATTR_TEMPERATURE, CONF_DEVICE, CONF_NAME,
+                                 PRECISION_HALVES, PRECISION_TENTHS,
+                                 TEMP_CELSIUS, PRECISION_WHOLE)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['pyotgw==0.1b0']
+
+CONF_FLOOR_TEMP = "floor_temperature"
+CONF_PRECISION = 'precision'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_DEVICE): cv.string,
+    vol.Optional(CONF_NAME, default="OpenTherm Gateway"): cv.string,
+    vol.Optional(CONF_PRECISION): vol.In([PRECISION_TENTHS, PRECISION_HALVES,
+                                          PRECISION_WHOLE]),
+    vol.Optional(CONF_FLOOR_TEMP, default=False): cv.boolean,
+})
+
+SUPPORT_FLAGS = (SUPPORT_TARGET_TEMPERATURE)
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Set up the opentherm_gw device."""
+    gateway = OpenThermGateway(config)
+    async_add_entities([gateway])
+
+
+class OpenThermGateway(ClimateDevice):
+    """Representation of a climate device."""
+
+    def __init__(self, config):
+        """Initialize the sensor."""
+        import pyotgw
+        self.pyotgw = pyotgw
+        self.gateway = self.pyotgw.pyotgw()
+        self._device = config.get(CONF_DEVICE)
```

Comment:
Use `dict[key]` for required config keys.

Reply:
Is this a Home Assistant specific requirement? I don't mind making the change, but a quick test confirmed that the performance impact is negligible.

Reply:
It's not about performance. We should use appropriate tools for each task. When we know that a key is in a dictionary we access the value with `dict[key]`. If we would be mistaken, the code would blow up, and we would know about our mistake.

Reply:
Fair enough, addressed in 105f8eb668b11445d0de0862f32aebb783b9e3c1